src/useResizeDetector.ts

```diff
@@ -1,36 +1,34 @@
-import { useLayoutEffect, useState, useRef, MutableRefObject } from 'react';
+import { useLayoutEffect, useState, useRef, MutableRefObject, useCallback } from 'react';
 
 import { patchResizeHandler, createNotifier, isSSR, patchResizeHandlerType } from './utils';
 
 import { Props, ReactResizeDetectorDimensions } from './ResizeDetector';
 
 type resizeHandlerType = MutableRefObject<null | patchResizeHandlerType>;
+interface RefCallback {
+  (node: null | Element): void;
+} 
 interface returnType<RefType> extends ReactResizeDetectorDimensions {
-  ref: MutableRefObject<null | RefType>;
+  ref: RefCallback;
 }
 
-interface FunctionProps<RefType> extends Props {
-  targetRef?: MutableRefObject<null | RefType>;
-}
-
-function useResizeDetector<RefType extends Element = Element>(props: FunctionProps<RefType> = {}): returnType<RefType> {
+const useResizeDetector =<RefType extends Element = Element>(props: Props = {}): returnType<RefCallback> => {
   const {
     skipOnMount = false,
     refreshMode,
     refreshRate = 1000,
     refreshOptions,
     handleWidth = true,
     handleHeight = true,
-    targetRef,
     observerOptions,
     onResize
   } = props;
 
   const skipResize: MutableRefObject<null | boolean> = useRef(skipOnMount);
```

Comment:
I was wandering about the use of skipResize here.
Why use a mutable ref object which would not trigger updates if its changing -?
I would have expected something like:
~ `const [skipResize, setSkipResize] = useState(skipOnMount)`

Reply:
Yes, I think you're right.

What I wanted to achieve is to get rid of unnecessary component rerender.   
`setSkipResize(true)` triggers a rerender, but `skipResize.current = true` - doesn't trigger any side effects.

But I agree that your approach is more clear and that rerender doesn't run any additional logic from inside the module. So it would be great if you can make a PR to update this.

Thanks in advance 👍

Reply:
Ok, I'll have a look if the code will benefit from a refactoring here. (But it's not critical for the usage here)